File: Code In Go Language/Fan.go
package main

type Fan struct {
	currentState State
}

func (object *Fan) Initialize() {
	object.currentState = new(Off)
}

func (object *Fan) setState(s State) {
	object.currentState = s
}

func (object *Fan) pull() {
	object.currentState.pull(object)
}

File: Code In C# Language/State2/Fan.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace State2
{
    class Fan
    {

        //its just a state wrapper

        private State currentState;

        public Fan()
        {
            currentState = new Off();
        }

        public void setState(State s)
        {
            currentState = s;
        }


        public void pull()
        {
            currentState.pull(this);
        }

    }
}
